src/clients/c/tb_client/context.zig

```diff
@@ -237,17 +249,18 @@ pub fn ContextType(
 
         /// Only one thread calls `deinit()`.
         /// Since it frees the Context, any further interaction is undefined behavior.
-        pub fn deinit(self: *Context) void {
+        pub fn deinit(self: *Context, root_allocator: std.mem.Allocator) void {
             self.signal.stop();
             self.thread.join();
             self.io.cancel_all();
 
             self.signal.deinit();
-            self.client.deinit(self.allocator);
-            self.message_pool.deinit(self.allocator);
+            self.client.deinit(self.gpa.allocator());
+            self.message_pool.deinit(self.gpa.allocator());
             self.io.deinit();
 
-            self.allocator.destroy(self);
+            assert(self.gpa.deinit() == .ok);
+            root_allocator.destroy(self);
```

Comment:
We also don't need to pass `root_allocator` to `deinit`, now we can use the GPA for dealocating the context (and assert it).

```suggestion
            var gpa: GPA = self.gpa;
            const allocator = gpa.allocator();
            allocator.destroy(self);
            assert(gpa.deinit() == .ok);
```